.rhdh/scripts/mirror-plugins.sh

```diff
@@ -0,0 +1,1215 @@
+#!/bin/bash
+#
+# Script to mirror Red Hat Developer Hub dynamic plugin OCI artifacts for deployments in restricted environments.
+# This script is installation-method agnostic and works with both operator and helm deployments
+# on both OpenShift and Kubernetes platforms.
+#
+# Requires: skopeo, tar, jq, podman
+
+set -euo pipefail
+
+NC='\033[0m'
+
+DEBUG=0
+
+PLUGIN_INDEX=""
+PLUGIN_LIST_FILE=""
+PLUGIN_URLS=()
+PLUGIN_IMAGES=()
+
+TO_REGISTRY=""
+TO_DIR=""
+FROM_DIR=""
+
+function logf() {
+  set -euo pipefail
+
+  local prefix=$1
+  local color=$2
+  local msg=$3
+  local fullMsg="[${prefix}] ${msg}"
+
+  if [[ "$TERM" == *"color"* ]]; then
+    echo -e "${color}${fullMsg}${NC}"
+  else
+    echo -e "${fullMsg}"
+  fi
+}
+
+function infof() {
+  logf "INFO" "${NC}" "$1"
+}
+
+function warnf() {
+  logf "WARN" "\033[0;33m" "$1"
+}
+
+function debugf() {
+  if [[ $DEBUG -eq 1 ]]; then logf "DEBUG" "\033[0;90m" "$1"; fi
+}
+
+function errorf() {
+  logf "ERROR" "\033[0;31m" "$1"
+}
+
+function check_tool() {
+  if ! command -v "$1" >/dev/null; then
+    echo "Error: Required tool '$1' is not installed." >&2
+    exit 1
+  fi
+}
+
+check_tool "skopeo"
```

Comment:
do any of these have VERSION requirements? What versions did you use, and should we set minimums so it's more clear what a customer needs to have installed? Alternatively could we provide a container image in which to run this script where the deps are already present?